pandas/_typing.py

```diff
@@ -118,6 +118,7 @@
     Concatenate: Any = None
 
 HashableT = TypeVar("HashableT", bound=Hashable)
+HashableT2 = TypeVar("HashableT2", bound=Hashable)
```

Comment:
Curious why we need a second one?

Reply:
needed for `_from_nested_dict`. When using the same TypeVar in a function signature, the type checkers bind these to the same type:

```py
def (x: HashableT) -> HashableT: ...

x(None) # return None
x((1, 2, 3)) # return tuple[int, int, int]
```

When we have a function that tries to bind two different sets of types, we need a new TypeVar

`def (x: HashableT, y: HashableT2) -> dict[HashableT, dict[HashableT2, Any]]: ...`

(pandas-stubs uses the name HashableT2)